cmd/resim/commands/sweep.go

```diff
@@ -0,0 +1,339 @@
+		if err != nil {
+			log.Fatal("failed to parse grid search config file: ", err)
+		}
+	} else {
+		// Parse the parameter name
+		parameterName := viper.GetString(sweepParameterNameKey)
+
+		// Parse the parameter values
+		parameterValues := viper.GetStringSlice(sweepParameterValuesKey)
+		if len(parameterValues) == 0 {
+			log.Fatal("failed to create sweep: you must specify at least one parameter value")
+		}
+
+		sweepParameters = append(sweepParameters, api.SweepParameter{
+			Name:   &parameterName,
+			Values: &parameterValues,
+		})
+	}
+
+	if !sweepGithub {
+		fmt.Println("Sweep parameters:")
+		for _, parameter := range sweepParameters {
+			fmt.Printf("  %s: %s\n", *parameter.Name, *parameter.Values)
+		}
+	}
+
+	// Parse the build ID
+	buildID, err := uuid.Parse(viper.GetString(sweepBuildIDKey))
+	if err != nil || buildID == uuid.Nil {
+		log.Fatal("failed to parse build ID: ", err)
+	}
+
+	var allExperienceIDs []uuid.UUID
+	var allExperienceNames []string
+
+	// Parse --experiences into either IDs or names
+	if viper.IsSet(sweepExperiencesKey) {
+		experienceIDs, experienceNames := parseUUIDsAndNames(viper.GetString(sweepExperiencesKey))
+		allExperienceIDs = append(allExperienceIDs, experienceIDs...)
+		allExperienceNames = append(allExperienceNames, experienceNames...)
+	}
+
+	metricsBuildID := uuid.Nil
+	if viper.IsSet(sweepMetricsBuildKey) {
+		metricsBuildID, err = uuid.Parse(viper.GetString(sweepMetricsBuildKey))
+		if err != nil {
+			log.Fatal("failed to parse metrics-build ID: ", err)
+		}
+	}
+
+	var allExperienceTagIDs []uuid.UUID
+	var allExperienceTagNames []string
+
+	// Parse --experience-tags
+	if viper.IsSet(sweepExperienceTagsKey) {
+		experienceTagIDs, experienceTagNames := parseUUIDsAndNames(viper.GetString(sweepExperienceTagsKey))
+		allExperienceTagIDs = append(allExperienceTagIDs, experienceTagIDs...)
+		allExperienceTagNames = append(allExperienceTagNames, experienceTagNames...)
+	}
+
+	// Build the request body
+	body := api.CreateParameterSweepJSONRequestBody{
+		BuildID:    &buildID,
+		Parameters: &sweepParameters,
+	}
+
+	if allExperienceIDs != nil {
+		body.ExperienceIDs = &allExperienceIDs
+	}
+
+	if allExperienceNames != nil {
+		body.ExperienceNames = &allExperienceNames
+	}
+
+	if allExperienceTagIDs != nil {
+		body.ExperienceTagIDs = &allExperienceTagIDs
+	}
+
+	if allExperienceTagNames != nil {
+		body.ExperienceTagNames = &allExperienceTagNames
+	}
+
+	if metricsBuildID != uuid.Nil {
+		body.MetricsBuildID = &metricsBuildID
+	}
+
+	// Make the request
+	response, err := Client.CreateParameterSweepWithResponse(context.Background(), body)
+	if err != nil {
+		log.Fatal("failed to create sweep:", err)
+	}
+	ValidateResponse(http.StatusCreated, "failed to create sweep", response.HTTPResponse)
+
+	if response.JSON201 == nil {
+		log.Fatal("empty response")
+	}
+	sweep := *response.JSON201
+
+	if !sweepGithub {
+		// Report the results back to the user
+		fmt.Println("Created sweep successfully!")
```

Comment:
This could be moved into the appropriate `if !sweepGitHub {` branch below

Reply:
Actually all of these Printlns could be moved into a single block after the checks, right?